analyzer-service/calculations.py

```diff
@@ -0,0 +1,169 @@
+'''dependencies'''
+import numpy as np
+import pandas as pd
+from scipy.stats import norm
+import statsmodels.api as sm
+from datetime import datetime
+
+from DBInterface import DBInterface
+
+def filterData(tData):
+  df = pd.DataFrame.from_dict(tData)
+  df = df.T
+  df.reset_index(level=0, inplace=True)
+  x = df.loc[:,'index']
+  current_date = datetime.today()
+  filterDate = (current_date - pd.Timedelta(weeks=26)) - pd.Timedelta(days=1)
+
+  for i in range(len(df.loc[:, 'index'])):
+    df.loc[i,'index'] = datetime.strptime(df.loc[i,'index'] , '%Y-%m-%d')
+
+  return df.loc[df['index'] > filterDate]
+
+
+def sixMonthReturn(df):
+  i = len(df.loc[:, 'index']) - 1
+  change = pd.to_numeric(df.iloc[i, 5]) -  pd.to_numeric(df.iloc[0, 5])
+  pctChange = change / pd.to_numeric(df.iloc[0, 5])
+  return round(pctChange, 4)
+
+
+def loadHistorical(ticker):
+  '''
+  load historical data for class attribute - ticker
+
+  @return bool set to true if functions completes
+  '''
+  client, collection = DBInterface.connect() #need try catch here
+  cursor = collection.find_one({"MetaData.2Symbol" : "{}".format(ticker)})
+  tName = cursor['MetaData']['2Symbol']
+  tData = cursor['TimeSeries(Daily)']
+
+  return (tName, tData)
+
+
+def calculateVar(tName, tData):
+  '''
+  Function to calculate the Value at Risk using the Variance/Covaraiance method
+
+  @return dict containing ticker symbol and 3 levels of value at risk (90, 95, 99)
+  '''
+  # make data into a pandas data frame
+  df = filterData(tData)
+  df = df.iloc[::-1]
+
+  # calculate percent returns per day
+  pReturns = pd.to_numeric(df.iloc[:, 5]).pct_change()
+
+  # calculate the percent return's mean and s.d.
+  pr_mean = np.mean(pReturns)
+  pr_std = np.std(pReturns)
+
+  # use normal distribution percentile function to get Value at Risk (90, 95, 99)
+  VaR_90 = norm.ppf(.1, pr_mean, pr_std)
+  VaR_95 = norm.ppf(.05, pr_mean, pr_std)
+  VaR_99 = norm.ppf(.01, pr_mean, pr_std)
```

Comment:
Thank you for doing these calculation at different levels

Reply:
No problem